cmd/manifest/init.go

```diff
@@ -0,0 +1,330 @@
+	cmd.Flags().StringVarP(&opts.DevPath, "file", "f", utils.DefaultManifest, "path to the manifest file")
+	cmd.Flags().BoolVarP(&opts.Overwrite, "replace", "r", false, "overwrite existing manifest file")
+	return cmd
+}
+
+// Init initializes a new okteto manifest
+func (mc *ManifestCommand) Init(ctx context.Context, opts *InitOpts) error {
+	if err := validateDevPath(opts.DevPath, opts.Overwrite); err != nil {
+		return err
+	}
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	composeFiles := utils.GetStackFiles(cwd)
+
+	var manifest *model.Manifest
+	if len(composeFiles) > 0 {
+		composePath, create, err := selectComposeFile(composeFiles)
+		if err != nil {
+			return err
+		}
+		if composePath != "" {
+			manifest, err = createFromCompose(composePath)
+			if err != nil {
+				return err
+			}
+		} else if create {
+			manifest, err = createNewCompose()
+			if err != nil {
+				return err
+			}
+		} else {
+			manifest, err = createFromKubernetes(cwd)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		selection, err := utils.AskForOptions([]string{"compose", "kubernetes"}, "How do you want to deploy your application?")
+		if err != nil {
+			return err
+		}
+		if selection == "compose" {
+			manifest, err = createNewCompose()
+			if err != nil {
+				return err
+			}
+		} else {
+			manifest, err = createFromKubernetes(cwd)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if manifest != nil {
+		mc.manifest = manifest
+		manifest.Name = utils.InferApplicationName(cwd)
+		if err := manifest.WriteToFile(opts.DevPath); err != nil {
+			return err
+		}
+		if opts.ShowCTA {
+			answer, err := utils.AskYesNo("Do you want to deploy?")
+			if err != nil {
+				return err
+			}
+			if answer {
+				kubeconfig := deploy.NewKubeConfig()
+				proxy, err := deploy.NewProxy(manifest.Name, kubeconfig)
+				if err != nil {
+					return err
+				}
+
+				c := &deploy.DeployCommand{
+					GetManifest:        mc.getManifest,
+					Kubeconfig:         kubeconfig,
+					Executor:           executor.NewExecutor(oktetoLog.GetOutputFormat()),
+					Proxy:              proxy,
+					TempKubeconfigFile: deploy.GetTempKubeConfigFile(manifest.Name),
+					K8sClientProvider:  okteto.NewK8sClientProvider(),
+				}
+
+				return c.RunDeploy(ctx, &deploy.Options{
+					Name:         manifest.Name,
+					ManifestPath: opts.DevPath,
+					Timeout:      5 * time.Minute,
+					Build:        true,
+				})
+			}
+			oktetoLog.Success("Okteto manifest configured correctly.\n    Run `okteto up` to activate your development container")
+		}
+	}
+	return nil
+}
+
+func validateDevPath(devPath string, overwrite bool) error {
+	if !overwrite && model.FileExists(devPath) {
+		return fmt.Errorf("%s already exists. Run this command again with the '--replace' flag to overwrite it", devPath)
+	}
+	return nil
+}
+
+func createFromCompose(composePath string) (*model.Manifest, error) {
```

Comment:
before creating from compose, we should ask for confirmation telling that the okteto manifest is optional if you have a compose file